### ISSUE TITLE
OnAuthRequired triggered during token AutoRenew

### ISSUE THREAD

**Issue by polklabs** [okta/okta-auth-js#729]
## Behavior 

When an IdToken expires the auto renew process is started. During this process OnAuthRequired is triggered causing the user to be redirect to the login page. Since they still have an active Okta session they are immediatly redirected back to where they were but may lose their current state.

OnAuthRequired function being used:
```
export function onAuthRequired(_oktaAuth: OktaAuthService, injector: Injector): void {
    console.log('On Auth Required');

    // Okta URI doesn't handle query params so set the redirect URL to the full path
    // and we can redirect them there after they login
    const loginHelperService: LoginHelperService = injector.get(LoginHelperService);
    loginHelperService.setRedirectUrl(location.pathname + location.search);

    _oktaAuth.signInWithRedirect({
      originalUri: location.pathname
    });
}
```

Example route implementation:
```
{
        path: 'home',
        component: HomeComponent,
        canActivate: [ ...oktaAuthGuard ],
        data: {title: 'Home', onAuthRequired }
  },
```

Same issue also occures when onAuthRequired is included directly in the Okta config. I also tried to set autoRenew to false in the oktaConfig but it didn't appear to do anything as auto renew was triggered anyway.

I can see the authorize endpoint being called with `prompt=none` but then immediately after triggers the authorize with a redirect.

Is there a way to tell if Okta Auth is in the middle of a token renew? With the new changes in version 5 there is a lot of conflicting and possibly missing information in the docs

## Expected Behavior

AutoRenew silently renews the IdToken in the background without redirecting or triggering OnAuthRequired

## Versions

@okta/okta-angular: ^3.1.0
@okta/okta-auth-js: ^5.0.0

Angular CLI: 10.2.3
Angular: 10.2.5
Node: 12.19.0

**Comment by bryanapellanes-okta** [comment]
Internal Reference: OKTA-394440

**Comment by polklabs** [comment]
Downgraded to version 4.9.0 and no longer experiencing the issue. Thanks

**Comment by polklabs** [comment]
I didn't get the refresh behavior after reverting to 4.9.0 so I thought I was fixed. Tested it again twice this morning and both times it was the same behavior as my original issue. It refreshed the page when it should have just refreshed the tokens.

I even tried removing the onAuthRequired method, allowing it to use the built-in and it had the same behavior.

When the token renew is triggered I see the following calls in the Network tab. The first two use prompt=none but the third redirects, I assume after onAuthRequired is called
 - https://\*\*\*\*.oktapreview.com/oauth2/v1/authorize?client_id=\*\*\*\*&code_challenge=\*\*\*\*&code_challenge_method=S256&nonce=\*\*\*\*&prompt=none&redirect_uri=http%3A%2F%2Flocalhost%3A4200%2Fimplicit%2Fcallback&response_mode=okta_post_message&response_type=code&state=\*\*\*\*&scope=profile%20openid
 - https://\*\*\*\*.oktapreview.com/oauth2/v1/token
 - https://\*\*\*\*.oktapreview.com/oauth2/v1/authorize?client_id=\*\*\*\*&code_challenge=\*\*\*\*&code_challenge_method=S256&nonce=\*\*\*\*&prompt=none&redirect_uri=http%3A%2F%2Flocalhost%3A4200%2Fimplicit%2Fcallback&response_mode=okta_post_message&response_type=code&state=\*\*\*\*&scope=profile%20openid
 - https://\*\*\*\*.oktapreview.com/oauth2/v1/token
 - https://\*\*\*\*.oktapreview.com/oauth2/v1/authorize?client_id=\*\*\*\*&code_challenge=\*\*\*\*&code_challenge_method=S256&nonce=\*\*\*\*&redirect_uri=http%3A%2F%2Flocalhost%3A4200%2Fimplicit%2Fcallback&response_type=code&state=\*\*\*\*&scope=profile%20openid
 - http://localhost:4200/implicit/callback?code=****&state=****

**Comment by polklabs** [comment]
@aarongranick-okta I changed `expireEarlySeconds` to 3300 for testing and that appeared to mostly fix the problem but after running for a couple hours it's now back to triggering onAuthRequired and refreshing the page every time

**Comment by aarongranick-okta** [comment]
@polklabs Is the issue you are seeing here captured by this new issue: https://github.com/okta/okta-auth-js/issues/732

**Comment by polklabs** [comment]
@aarongranick-okta It is the same behavior but It may not be the same cause since v5.0.0 doesn't set the token to **null** in `updateAuthState()`.
I'll have to test it again with the event watchers I added. But maybe it's something that will already be fixed in the next okta-angular version.

That last 2 comments I added here are captured by issue #732

**Comment by polklabs** [comment]
@aarongranick-okta Just tested it againts okta-auth-js 5.0.0 and it is the same exact issue as #732 only in this case the token is being set to null in `isAuthenticated()`
https://github.com/okta/okta-auth-js/blob/5.0/lib/OktaAuth.ts#L497

**Comment by shuowu-okta** [comment]
This issue is fixed in [5.2.3](https://github.com/okta/okta-auth-js/releases/tag/okta-auth-js-5.2.3) patch. Closing this issue, feel free to re-open if the issue still exist.
